### DISCUSSION TITLE
GPG signing from codespaces consistently failing

### DISCUSSION THREAD

**Discussion by sbilstein** [community/community#29556] [Codespaces]
Hey folks, I've been using the codesigning feature on Github codespaces successfully for a few months but recently I cannot commit anymore.

I have GPG verification turned on and my credentials shared to my org's repository, codesigning is enabled on our codespaces (the other devs on my team have no issue.)

More recently though, I get this error quite consistently when committing:
```
error: gpg failed to sign the data
fatal: failed to write commit object
 ```
Here is the trace:
```
17:58:39.109986 git.c:444               trace: built-in: git commit -m test
17:58:39.115991 run-command.c:664       trace: run_command: GIT_EDITOR=: GIT_INDEX_FILE=.git/index .husky/pre-commit
→ No staged files match any configured task.
→ No staged files match any configured task.
→ No staged files match any configured task.
17:58:41.194693 run-command.c:664       trace: run_command: /.codespaces/bin/gh-gpgsign --status-fd=2 -bsau 'GitHub <noreply@github.com>'
error: gpg failed to sign the data
fatal: failed to write commit object
```

**Comment by soumyamahunt** [comment]
I am also getting similar error:
> 17:01:04.070082 run-command.c:659       trace: run_command: /.codespaces/bin/gh-gpgsign --status-fd=2 -bsau 'GitHub <noreply@github.com>'
error: gpg failed to sign the data:
[GNUPG:] BEGIN_SIGNING
2023/09/20 17:01:04 error signing commit: error signing commit: error making request: 403 | Author is invalid, error making request: 403 | Author is invalid<br/>
fatal: failed to write commit object
error: could not commit staged changes.

Did you find any fix for this?

**Reply by sbilstein** [reply]
@soumyamahunt no...I had to delete and recreate codespaces until it started working randomly

**Reply by soumyamahunt** [reply]
Thanks @sbilstein. I think this issue happens when rebasing, that too randomly.

**Reply by n2taylor** [reply]
I've seen this after rebasing as well. Killing your terminal and opening a new one typically fixes it.

**Comment by IndianPhotoAcademy** [comment]
Verify that your Codespace has your GPG key configured correctly. To confirm, run gpg --list-secret-keys. Use git config --global user.signingkey <KEY_ID> to set your Git configuration after that. When using SSH forwarding, make sure the keys are being forwarded correctly by the gpg-agent. Verify that Pinentry is installed and available as well.
